### PR TITLE
SPEC-1292: Add rule to ignore read preference for standalone servers

### PR DESCRIPTION
**Pull request by jyemin**
https://jira.mongodb.org/browse/SPEC-1292

I'm open to adding a prose test for this, but currently there is nothing covering the existing behavior.

### CODE REVIEW

File: source/server-selection/server-selection.rst [hunk 1]
```diff
@@ -885,6 +885,8 @@ to the server differently:
 - Type Mongos: the read preference is sent to the server using the rules
   for `Passing read preference to mongos`_.
 
+- Type Standalone: the read preference is ignored
```
Comment (xdg) [inline source/server-selection/server-selection.rst]: Each time I've considered this spec change, I get worried about SlaveOK, and then remember that we're talking about Standalone and not Single.  But it makes me wonder if we should add something for clarity like: `(note: this does not apply to Topology Type Single for other server types than Standalone)`

Comment (jmikola) [reply source/server-selection/server-selection.rst]: That note seems like a reasonable way to preemptively avoid a misunderstanding.

I would also suggest rewording "the read preference is ignored" as "the read preference is never sent to the server" or "drivers MUST NOT send the read preference to the server" if we want to be prescriptive.

In the context of this section ("the read preference is communicated to the server"), "is ignored" makes sense but a rewording would make this clearer both here and in the changelog.

File: source/server-selection/server-selection.rst [hunk 2]
```diff
@@ -1684,6 +1686,8 @@ References
 Changes
 =======
 
+2019-05-20: Added rule to ignore read preference for standalone servers
```
Comment (jmikola) [inline source/server-selection/server-selection.rst]: "to never send read preference for standalone servers". Optionally: "standalone server types" to avoid any ambiguity about topology/server types.

File: source/server-selection/server-selection.rst (hunk 1 above)
Comment (jyemin) [reply source/server-selection/server-selection.rst]: @xdg, if you look at this change in context I think the meaning is clear, as it appears in a section titled `Topology type: Single", and there are bullets that cover all other server types.  In that context, the suggested note sounds a bit weird.  

@jmikola, I agree with your suggestion and will make it in the next patch, but I'll wait to hear back from @xdg first.

Comment (xdg) [reply source/server-selection/server-selection.rst]: Looking with more diff context, I agree that my additional language isn't needed, but I also agree with Jeremy about avoiding the term "ignored".

Comment (jyemin) [reply source/server-selection/server-selection.rst]: Done

File: source/server-selection/server-selection.rst (hunk 2 above)
Comment (jyemin) [reply source/server-selection/server-selection.rst]: Done
